src/main/java/com/zufar/icedlatte/order/api/OrderProvider.java

```diff
@@ -0,0 +1,89 @@
+package com.zufar.icedlatte.order.api;
+
+import com.zufar.icedlatte.openapi.dto.*;
+import com.zufar.icedlatte.order.converter.OrderDtoConverter;
+import com.zufar.icedlatte.order.entity.Order;
+import com.zufar.icedlatte.order.entity.OrderItem;
+import com.zufar.icedlatte.order.repository.OrderRepository;
+import com.zufar.icedlatte.product.repository.ProductInfoRepository;
+import com.zufar.icedlatte.security.api.SecurityPrincipalProvider;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.zufar.icedlatte.order.api.Calculator.calculate;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class OrderProvider {
+
+    private final OrderRepository orderRepository;
+    private final ProductInfoRepository productInfoRepository;
+    private final OrderDtoConverter orderDtoConverter;
+    private final SecurityPrincipalProvider securityPrincipalProvider;
+    public static final int DEFAULT_PRODUCTS_QUANTITY = 0;
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED)
+    public AddedOrder addNewOrder(final OrderDto orderBody) {
+        UUID userId = securityPrincipalProvider.getUserId();
+        log.info("User id: {}", userId);
+        var order = createNewOrder(orderBody, userId);
+        orderRepository.save(order);
+        log.info("New order was created and saved to database.");
+        return orderDtoConverter.toAddedOrder(order);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED)
+    public List<AddedOrder> getOrdersByStatus(final List<OrderStatus> statusList) {
+        var userId = securityPrincipalProvider.getUserId();
+        log.info("User id: {}", userId);
+        // TODO: should I create a custom SQL query, rather than filter here?
+        var ordersStream = orderRepository.findAllByUserId(userId).stream();
+        if (statusList != null) {
+            var set = new HashSet<>(statusList);
+            ordersStream = ordersStream.filter(order -> set.contains(order.getStatus()));
+        }
+        return ordersStream.map(orderDtoConverter::toAddedOrder).toList();
+    }
+
+    private Order createNewOrder(OrderDto orderBody, UUID userId) {
```

Comment:
put it to another service OrderCreator

Reply:
Done in in [46e65b4](https://github.com/Sunagatov/Iced-Latte/pull/217/commits/46e65b4c9646349a131383ace42afdeddba4b663)